engine/src/engine.ts

```diff
@@ -118,9 +118,11 @@ export function createMoveToShow(move: string, p: PlayerData, executeMove: () =>
 
   const formerBooster = p.tiles.booster;
 
-  executeMove();
-
-  p.off("move-tokens", listener);
+  try {
```

Comment:
I'd have moved `p.on(...)` just above the `try` (I'm a bit paranoid)

Reply:
I agree - it's a good pattern (will change later)